src/test/java/com/mathworks/ci/MatlabBuilderTest.java

```diff
@@ -87,9 +88,17 @@ public void testTearDown() {
 
     private String getMatlabroot(String version) throws URISyntaxException {
         ClassLoader classLoader = MatlabBuilderTest.class.getClassLoader();
-        String matlabRoot = new File(
-                classLoader.getResource("versioninfo/" + version + "/VersionInfo.xml").toURI())
-                        .getAbsolutePath().replace(FileSeperator + "VersionInfo.xml", "");
+        String matlabRoot;
```

Comment:
This bit of code is kind of hard to follow.  Could you assign the result of `classLoader.getResource(...)` to a variable and then use that in the if condition?

```java
String matlabRoot;
URL versionInfo = classLoader.getResource("versioninfo/"+version+"/VersionInfo.xml");

if(versionInfo!=null){
    matlabRoot = new File(versionInfo.toURI()).getAbsolutePath();
}else{
    matlabRoot = new File(classLoader.getResource("versioninfo/R2017a/VersionInfo.xml").toURI()).getAbsolutePath().replace("R2017a",version);
}

return matlabRoot.replace(FileSeparator+"VersionInfo.xml","");
```
Not everyone is a fan of Optionals, but I could also see the code like this:

```java
private String getMatlabroot(String version) throws URISyntaxException {
     String defaultVersionInfo = "versioninfo/R2017a/VersionInfo.xml";
     String userVersionInfo = "versioninfo/"+version+"/VersionInfo.xml";

      Optional<URL> matlabRootURL = Optional.ofNullable(getResource(userVersionInfo)).orElseGet(() -> getResource(defaultVersionInfo));
      
      File matlabRoot = new File(matlabRootURL.get().toURI());

      return matlabRoot.getAbsolutePath().replace(FileSeperator+"VersionInfo.xml","").replace("R2017a",version);      
}

private URL getResource(String resource){
    return MatlabBuilderTest.class.getClassLoader().getResource(resource);   
}
```

Reply:
Thanks @jwilkick-mw  . it was a good suggestion for using Optional <T> class